### ISSUE TITLE
[Issue]: CUDA FP16 test passed, but generated images are gray

### ISSUE THREAD

**Issue by Mantas-2155X** [vladmandic/automatic#900]
### Issue Description

Running daf90cb and checking the logs, my system should support FP16 but the generated images are all gray. 
Forcing FP32 with `no-half` resolves the issue.

This is a known problem on RX 5000 series cards. However, even using `upcast sampling` does not resolve it and the generated images are still gray when `no-half` is unselected.

Prompt (gray at any prompt and sampling method):
`1girl, smile, blonde hair` with `Euler a`, seed  `3599624310`

Generated Image:
![00004-3599624310-1girl, smile, blonde hair](https://github.com/vladmandic/automatic/assets/13873974/3507db46-0413-4e84-9d41-906047731ad1)

Logs:
[setup.log](https://github.com/vladmandic/automatic/files/11467086/setup.log)

### Application
Version: `daf90cb6 Fri May 12 14:23:51 2023 -0400`
Torch: `1.13.1+rocm5.2`

### System
Kernel: `Linux 6.2.6-tkg-pds`
Browser: `Firefox Nightly 114.0a1`
GPU: `AMD Radeon RX 5700 XT`

### Notes
I had to downgrade Torch to 1.13.1 because with Torch 2 the model never loads. Will make an issue about that later.

**Comment by vladmandic** [comment]
this is all good-to-know, but not sure what should i do about it?
fp16 test i wrote is a simple one that fails on most common operation which is 
> layerNorm = torch.nn.LayerNorm(4, eps=0.00001, elementwise_affine=True, dtype=torch.float16, device=device)

its quite possible (and likely) that your gpu fails some other test - but there is no single test i could do.
if you can find out which ops fails on your gpu, i can add it to my auto-test-on-startup.

**Comment by Mantas-2155X** [comment]
I see, unfortunately I don't have the knowledge to check what operations fail, thought there was simply an issue with the test. 
Thank you for looking into it

**Comment by vladmandic** [comment]
> I see, unfortunately I don't have the knowledge to check what operations fail, thought there was simply an issue with the test. Thank you for looking into it

I wish torch provided actual test instead of user trying what works and what doesn't.
